external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/KafkaTestUtils.scala

```diff
@@ -178,8 +178,9 @@ private[kafka010] class KafkaTestUtils extends Logging {
 
   /** Send the array of messages to the Kafka broker */
   def sendMessages(topic: String, messages: Array[String]): Unit = {
-    producer = new Producer[String, String](new ProducerConfig(producerConfiguration))
-    producer.send(messages.map { new KeyedMessage[String, String](topic, _ ) }: _*)
+    producer = new KafkaProducer[String, String](producerConfiguration)
+    val records = messages.map { new ProducerRecord[String, String](topic, _) }
+    records.map(producer.send)
```

Comment:
This type of thing is even essential for correctness in some cases, like https://issues.apache.org/jira/browse/SPARK-16664